addons/account/models/account_move.py

```diff
@@ -253,7 +253,16 @@ def _sequence_fixed_regex(self):
     made_sequence_hole = fields.Boolean(compute='_compute_made_sequence_hole')
     show_name_warning = fields.Boolean(store=False)
     type_name = fields.Char('Type Name', compute='_compute_type_name')
-    country_code = fields.Char(related='company_id.account_fiscal_country_id.code', readonly=True)
+    country_code = fields.Char(compute='_compute_country_code')
+
+    @api.depends('company_id', 'fiscal_position_id')
+    def _compute_country_code(self):
+        for move in self:
+            if move.fiscal_position_id.foreign_vat:
+                move.country_code = move.fiscal_position_id.country_id.code
+            else:
+                move.country_code = move.company_id.account_fiscal_country_id.code
```

Comment:
We should probably move this compute down in the file, together with the other computes (line 636+)

Reply:
Hmm, not sure why we should change the `country_code`  as we also have the `tax_country_code` where the `country_code` tells us which country the company is from and the `tax_country_code` in which country the move is eligible and liable in terms of taxes.